src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -241,4 +230,110 @@ private void emitGlobalSetter(Constant segmentConstant, Constant vhConstant, Str
             """);
         decrAlign();
     }
+
+    public String emitGlobalSegment(MemoryLayout layout, String javaName, String nativeName, Declaration declaration) {
```

Comment:
This is a recurrent theme: sometimes the "emit" method will return the name of a relevant variable that has been generated, so that other code can refer to it.